connector/signaltometrics/connector.go

```diff
@@ -0,0 +1,281 @@
+						dps := metric.Gauge().DataPoints()
+						for l := 0; l < dps.Len(); l++ {
+							dp := dps.At(l)
+							filteredDPAttrs := getFilteredAttributes(dp.Attributes(), md.Attributes)
+							if filteredDPAttrs.Len() != len(md.Attributes) {
+								// If all the configured attributes are not present in
+								// source metric then don't count them.
+								continue
+							}
+							multiError = errors.Join(multiError, aggregate(dp, filteredDPAttrs))
+						}
+					case pmetric.MetricTypeSum:
+						dps := metric.Sum().DataPoints()
+						for l := 0; l < dps.Len(); l++ {
+							dp := dps.At(l)
+							filteredDPAttrs := getFilteredAttributes(dp.Attributes(), md.Attributes)
+							if filteredDPAttrs.Len() != len(md.Attributes) {
+								// If all the configured attributes are not present in
+								// source metric then don't count them.
+								continue
+							}
+							multiError = errors.Join(multiError, aggregate(dp, filteredDPAttrs))
+						}
+					case pmetric.MetricTypeSummary:
+						dps := metric.Summary().DataPoints()
+						for l := 0; l < dps.Len(); l++ {
+							dp := dps.At(l)
+							filteredDPAttrs := getFilteredAttributes(dp.Attributes(), md.Attributes)
+							if filteredDPAttrs.Len() != len(md.Attributes) {
+								// If all the configured attributes are not present in
+								// source metric then don't count them.
+								continue
+							}
+							multiError = errors.Join(multiError, aggregate(dp, filteredDPAttrs))
+						}
+					case pmetric.MetricTypeHistogram:
+						dps := metric.Histogram().DataPoints()
+						for l := 0; l < dps.Len(); l++ {
+							dp := dps.At(l)
+							filteredDPAttrs := getFilteredAttributes(dp.Attributes(), md.Attributes)
+							if filteredDPAttrs.Len() != len(md.Attributes) {
+								// If all the configured attributes are not present in
+								// source metric then don't count them.
+								continue
+							}
+							multiError = errors.Join(multiError, aggregate(dp, filteredDPAttrs))
+						}
+					case pmetric.MetricTypeExponentialHistogram:
+						dps := metric.ExponentialHistogram().DataPoints()
+						for l := 0; l < dps.Len(); l++ {
+							dp := dps.At(l)
+							filteredDPAttrs := getFilteredAttributes(dp.Attributes(), md.Attributes)
+							if filteredDPAttrs.Len() != len(md.Attributes) {
+								// If all the configured attributes are not present in
+								// source metric then don't count them.
+								continue
+							}
+							multiError = errors.Join(multiError, aggregate(dp, filteredDPAttrs))
+						}
+					case pmetric.MetricTypeEmpty:
+						multiError = errors.Join(multiError, fmt.Errorf("metric %q: invalid metric type: %v", metric.Name(), metric.Type()))
+					}
+				}
+			}
+		}
+	}
+	if multiError != nil {
+		return multiError
+	}
+	aggregator.Finalize(sm.dpMetricDefs)
+	return sm.next.ConsumeMetrics(ctx, processedMetrics)
+}
+
+func (sm *signalToMetrics) ConsumeLogs(ctx context.Context, logs plog.Logs) error {
+	var multiError error
```

Comment:
Here and in other places, maybe check if there are metric definitions first, and short-circuit if there aren't.

Reply:
Fixed!